### PR TITLE
Clean all build logic builds before running perf tests on Windows

### PR DESCRIPTION
**Pull request by jjohannes**

### CODE REVIEW

**Comment by jjohannes** [comment]
https://builds.gradle.org/buildConfiguration/Gradle_Check_PerformanceTest6_Trigger?branch=jjohannes%2Fclean-build-logic&mode=builds

File: .teamcity/common/performance-test-extensions.kt [hunk 1]
```diff
@@ -75,22 +75,27 @@ fun BuildSteps.substDirOnWindows(os: Os, buildCache: BuildCache) {
...
-            workingDir = "P:/build-logic"
-            executionMode = BuildStep.ExecutionMode.ALWAYS
-            gradleWrapperPath = "../"
-            gradleParams = (
-                buildToolGradleParameters() +
-                    buildScanTag("PerformanceTest") +
-                    buildCache.gradleParameters(os)
-                ).joinToString(separator = " ")
-        }
+        cleanBuildLogicBuild("build-logic-commons", buildCache, os)
```
Comment (wolfs) [inline .teamcity/common/performance-test-extensions.kt:78]: This also needs to happen in `removeSubstDirOnWindows`.

**Review by wolfs — CHANGES_REQUESTED** [review]

File: .teamcity/common/performance-test-extensions.kt [hunk 2]
```diff
@@ -75,22 +75,27 @@ fun BuildSteps.substDirOnWindows(os: Os, buildCache: BuildCache) {
...
+    }
+}
+
+private fun BuildSteps.cleanBuildLogicBuild(buildName: String, buildCache: BuildCache, os: Os) {
+    // Gradle detects overlapping outputs when running first on a subst drive and then in the original location.
+    // Even when running clean builds on CI, we don't run clean in buildSrc, so there may be stale leftover files there.
+    // This means that we need to clean buildSrc before running for the first time on the subst drive
+    // and before running the first time on the original location again.
+    gradleWrapper {
+        name = "CLEAN_BUILD_SRC_ON_SUBST_DRIVE"
```
Comment (blindpirate) [inline .teamcity/common/performance-test-extensions.kt:89]:
```suggestion
        name = "CLEAN_${buildName.toUpperCase().replace('-', '_')}_ON_SUBST_DRIVE"
```

**Review by blindpirate — APPROVED** [review]
LGTM, I left a suggestion.

File: .teamcity/common/performance-test-extensions.kt (hunk 2 above)
Comment (jjohannes) [reply .teamcity/common/performance-test-extensions.kt:89]: Good catch. Did not think about that the things has a name at all. Will fix.

File: .teamcity/common/performance-test-extensions.kt (hunk 1 above)
Comment (jjohannes) [reply .teamcity/common/performance-test-extensions.kt:78]: Thanks. Will fix.

**Comment by jjohannes** [comment]
https://builds.gradle.org/buildConfiguration/Gradle_Check_PerformanceTest6_Trigger/40087112

**Comment by jjohannes** [comment]
PTAL @wolfs. I think it looks good now. For example: https://builds.gradle.org/buildConfiguration/Gradle_Check_PerformanceTest6bucket1/40087107

**Review by wolfs — APPROVED** [review]
LGTM! Thank you Jendrik!

Do you have an idea how we can make something fail when we have more build logic builds so we don't forget to add those clean steps in the future? It may be a good idea to extract the list of projects somewhere and then write a test which checks that we don't have more.

**Pull request merged**